### ISSUE TITLE
Crash in mono_get_seq_points

### ISSUE THREAD

**Issue by Therzok** [mono/mono#17545]
## Steps to Reproduce

1. N/A - https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1007554
2. See attached log for stacktrace information

### Current Behavior

From the looks of it, logging an exception is causing a crash.

### Expected Behavior

No crash.

## On which platforms did you notice this

[*] macOS
[ ] Linux
[ ] Windows

**Version Used**:

Mono 6.4.0.208 (2019-06/07c23f2ca43) (64-bit)

## Stacktrace

```
=================================================================
	Native Crash Reporting
=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Native stacktrace:
=================================================================
	0x10bd199eb - /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/libmonosgen-2.0.1.dylib : mono_dump_native_crash_info
	0x10bcb2875 - /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/libmonosgen-2.0.1.dylib : mono_handle_native_crash
	0x10bd13b77 - /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/libmonosgen-2.0.1.dylib : altstack_handle_and_restore
	0x10bf1f419 - /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/libmonosgen-2.0.1.dylib : monoeg_g_hash_table_lookup
	0x10bc11271 - /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/libmonosgen-2.0.1.dylib : mono_get_seq_points
	0x10bc11388 - /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/libmonosgen-2.0.1.dylib : mono_find_prev_seq_point_for_native_offset
	0x10bcaf421 - /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/libmonosgen-2.0.1.dylib : ves_icall_get_trace
	0x11715ce08 - Unknown
	0x10ffd97bb - Unknown
	0x11008df9b - Unknown
	0x135bb448b - Unknown
	0x135bb4beb - Unknown
	0x134c375b3 - Unknown
	0x116ff9018 - Unknown
	0x116ff8e83 - Unknown
	0x116fd6c0b - Unknown
	0x116ff89b3 - Unknown
	0x116ff1c1b - Unknown
	0x10bc0ecb9 - /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/libmonosgen-2.0.1.dylib : mono_jit_runtime_invoke
	0x10be2c324 - /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/libmonosgen-2.0.1.dylib : do_runtime_invoke
	0x10be56272 - /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/libmonosgen-2.0.1.dylib : worker_callback
	0x10bd88b7c - /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/libmonosgen-2.0.1.dylib : worker_thread
	0x10be53733 - /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/libmonosgen-2.0.1.dylib : start_wrapper
	0x7fff67106d76 - /usr/lib/system/libsystem_pthread.dylib : _pthread_start
	0x7fff671035d7 - /usr/lib/system/libsystem_pthread.dylib : thread_start

```

```
=================================================================
	Managed Stacktrace:
=================================================================
	  at <unknown> <0xffffffff>
	  at System.Diagnostics.StackTrace:get_trace <0x00087>
	  at System.Diagnostics.StackTrace:.ctor <0x00072>
	  at System.Diagnostics.StackTrace:.ctor <0x00052>
	  at System.Environment:GetStackTrace <0x000ca>
	  at System.Exception:GetStackTrace <0x00092>
	  at System.Exception:ToString <0x00232>
	  at System.Exception:ToString <0x00162>
	  at System.Exception:ToString <0x00162>
	  at System.Exception:ToString <0x0004a>
	  at System.AggregateException:ToString <0x00092>
	  at System.Text.StringBuilder:AppendFormatHelper <0x01196>
	  at System.String:FormatHelper <0x000ba>
	  at System.String:Format <0x0010a>
	  at MonoTouch.Design.LoggingService:Log <0x001aa>
	  at MonoTouch.Design.LoggingService:LogError <0x0018a>
	  at <>c__DisplayClass289_0`1:<EnsureSession>b__0 <0x01112>
	  at System.Threading.Tasks.Task`1:InnerInvoke <0x00097>
	  at System.Threading.Tasks.Task:Execute <0x00042>
	  at System.Threading.Tasks.Task:ExecutionContextCallback <0x00072>
	  at System.Threading.ExecutionContext:RunInternal <0x0022d>
	  at System.Threading.ExecutionContext:Run <0x0005a>
	  at System.Threading.Tasks.Task:ExecuteWithThreadLocal <0x00142>
	  at System.Threading.Tasks.Task:ExecuteEntry <0x00102>
	  at System.Threading.Tasks.Task:System.Threading.IThreadPoolWorkItem.ExecuteWorkItem <0x00042>
	  at System.Threading.ThreadPoolWorkQueue:Dispatch <0x00399>
	  at System.Threading._ThreadPoolWaitCallback:PerformWaitCallback <0x0007a>
	  at <Module>:runtime_invoke_bool <0x000a5>
=================================================================
```

**Comment by lambdageek** [comment]
Possibly related to https://github.com/mono/mono/issues/15518 or to #8033

@vargaz Could `domain_jit_info(domain)->seq_points`  be NULL or corrupted?

**Comment by lambdageek** [comment]
Looks like the same issue as #17737 which is fixed by #17738 (master) and #17741 (2019-10)

**Comment by Therzok** [comment]
The VSTS issue is targeting what would be the equivalent of 2019-08 in mono. Can we have this backported to 2019-08? cc @slluis
